dev/release/verify-release-candidate.sh

```diff
@@ -51,12 +51,19 @@ HERE=$(cd `dirname "${BASH_SOURCE[0]:-$0}"` && pwd)
 
 ARROW_DIST_URL='https://dist.apache.org/repos/dist/dev/arrow'
 
-: ${ARROW_HAVE_CUDA:=}
-if [ -z "$ARROW_HAVE_CUDA" ]; then
-  if nvidia-smi --list-gpus 2>&1 > /dev/null; then
-    ARROW_HAVE_CUDA=yes
+detect_cuda() {
+  if ! (which nvcc && which nvidia-smi) > /dev/null; then
+    return 1
   fi
+
+  local n_gpus=$(nvidia-smi --list-gpus | wc -l)
+  return $((${n_gpus} < 1))
```

Comment:
It seems this will return true if there is no GPU?
On my system:
```bash
$ nvidia-smi --list-gpus 
GPU 0: GeForce GT 730 (UUID: GPU-9659e3d1-ea83-8652-7a61-8bc8d7b7da2e)
$ n_gpus=$(nvidia-smi --list-gpus | wc -l)
$ echo $n_gpus 
1
$ echo $((${n_gpus} < 1))
0
```

Reply:
Don't forget that in bash, the `if` command expects a return code of 0 to denote success.

Reply:
Ouch.